File: src/main/java/ca/yyx/hu/extractor/SystemMediaExtractor.java
package ca.yyx.hu.extractor;

import android.content.Context;
import android.media.MediaCodec;
import android.media.MediaExtractor;
import android.media.MediaFormat;
import android.net.Uri;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.util.Map;

/**
 * @author algavris
 * @date 30/04/2016.
 */
public class SystemMediaExtractor implements MediaExtractorInterface {
    private final MediaExtractor mExtractor = new MediaExtractor();

    @Override
    public int readSampleData(ByteBuffer buffer, int offset) {
        return mExtractor.readSampleData(buffer, offset);
    }

    @Override
    public void getSampleCryptoInfo(MediaCodec.CryptoInfo cryptoInfo) {
        mExtractor.getSampleCryptoInfo(cryptoInfo);
    }

    @Override
    public void release() {
        mExtractor.release();
    }

    @Override
    public void setDataSource(byte[] content, int width, int height) {

    }


    @Override
    public void setDataSource(Context context, Uri uri, Map<String, String> headers) throws IOException {
        mExtractor.setDataSource(context, uri, headers);
    }

    @Override
    public int getTrackCount() {
        return mExtractor.getTrackCount();
    }

    @Override
    public void unselectTrack(int index) {
        mExtractor.unselectTrack(index);
    }

    @Override
    public MediaFormat getTrackFormat(int index) {
        return mExtractor.getTrackFormat(index);
    }

    @Override
    public void selectTrack(int index) {
        mExtractor.selectTrack(index);
    }

    @Override
    public int getSampleFlags() {
        return mExtractor.getSampleFlags();
    }

    @Override
    public long getSampleTime() {
        return mExtractor.getSampleTime();
    }

    @Override
    public void advance() {
        mExtractor.advance();
    }
}

File: app/src/main/java/ca/anodsplace/headunit/extractor/SystemMediaExtractor.kt
package ca.anodsplace.headunit.extractor

import android.content.Context
import android.media.MediaCodec
import android.media.MediaExtractor
import android.media.MediaFormat
import android.net.Uri

import java.io.IOException
import java.nio.ByteBuffer

/**
 * @author algavris
 * *
 * @date 30/04/2016.
 */
class SystemMediaExtractor : MediaExtractorInterface {
    private val mExtractor = MediaExtractor()

    override fun readSampleData(buffer: ByteBuffer, offset: Int): Int {
        return mExtractor.readSampleData(buffer, offset)
    }

    override fun getSampleCryptoInfo(cryptoInfo: MediaCodec.CryptoInfo) {
        mExtractor.getSampleCryptoInfo(cryptoInfo)
    }

    override fun release() {
        mExtractor.release()
    }

    override fun setDataSource(content: ByteArray, width: Int, height: Int) {

    }

    @Throws(IOException::class)
    override fun setDataSource(context: Context, uri: Uri, headers: Map<String, String>) {
        mExtractor.setDataSource(context, uri, headers)
    }

    override val trackCount: Int
        get() = mExtractor.trackCount

    override fun unselectTrack(index: Int) {
        mExtractor.unselectTrack(index)
    }

    override fun getTrackFormat(index: Int): MediaFormat {
        return mExtractor.getTrackFormat(index)
    }

    override fun selectTrack(index: Int) {
        mExtractor.selectTrack(index)
    }

    override val sampleFlags: Int
        get() = mExtractor.sampleFlags

    override val sampleTime: Long
        get() = mExtractor.sampleTime

    override fun advance() {
        mExtractor.advance()
    }
}
